### ISSUE TITLE
TST - CI fails when importing from ``benchmark_utils``

### ISSUE THREAD

**Issue by Badr-MOUFAD** [Badr-MOUFAD/benchmark_sqrt_lasso#4]
Throughout https://github.com/benchopt/benchmark_sqrt_lasso/pull/3,
There seemed to be an issue with importing from ``benchmark_utils`` folder in the CI test.

From the error log,
```shell
  File "benchmark_utils/__init__.py", line 2, in <module>
    from numba import njit
ModuleNotFoundError: No module named 'numba'
``` 

it appears that `benchopt` imports utils before installing the corresponding packages.
However, [benchmark bilevel](https://github.com/benchopt/benchmark_bilevel/blob/main/benchmark_utils/numba_utils.py) has the same case yet it successfully passes the CI tests.

**Comment by tomMoral** [comment]
Yes, to make it possible to create the `benchmark_utils` module, we need to be able to import the module whne launching the benchmark so `__init__.py` should always be importable. But for other files in `benchmark_utils`, you can do whatever you want.
